### DISCUSSION TITLE
customize dagster run_id to include now timestamp

### DISCUSSION THREAD

**Discussion by nibu147** [dagster-io/dagster#4045] [Q&A]
Is there a way customize dagster run_id to include now timestamp instead of having an alphanumerical string which might be difficult to read? thanks

**Comment by sryza** [comment]
~~Hi @nibu147 - customizing the run ID isn't possible at this time.~~

**Reply by sryza** [reply]
I was wrong about the above ^^. If you're launching pipelines via Dagster's GraphQL API, you can provide a run ID. It's not currently possible with our Python APIs, but we may add the ability to do it in the future.

**Reply by nibu147** [reply]
that would be a very nice feature! thank you for your answer

**Reply by ahoneycutt** [reply]
You can customize the `run_id` in python but you must use dagster >= 0.10.0 and `dagster.cli.pipeline.execute_launch_command` to launch your pipeline. This is how I inject run_id into my pipeline.

**Comment by ahoneycutt** [comment]
@nibu147 If you can change the way you are executing your pipeline, then I would suggest using the `dagster.cli.pipeline.execute_launch_command` function to launch your pipeline. You should be able to pass `run_id` in the `kwargs` parameter, e.g.
```python
result = execute_launch_command(DagsterInstance.get(), {'run_id': 'my-run-id', ...})
```
This approach will require dagster >= 0.10.0, however.
